llvm/tools/ir2builder/ir2builder.cpp

```diff
@@ -0,0 +1,1702 @@
+  } break;
+  case Instruction::Sub: {
+    call = getBinArithOp("Sub", op1, op2, I);
+  } break;
+  case Instruction::FSub: {
+    call = getFPBinArithOp("FSub", op1, op2, I);
+  } break;
+  case Instruction::Mul: {
+    call = getBinArithOp("Mul", op1, op2, I);
+  } break;
+  case Instruction::FMul: {
+    call = getFPBinArithOp("FMul", op1, op2, I);
+  } break;
+  case Instruction::UDiv: {
+    call = "CreateUDiv(" + op1 + ", " + op2 + ", \"\", " +
+           to_str(I->isExact()) + ")";
+  } break;
+  case Instruction::SDiv: {
+    call = "CreateSDiv(" + op1 + ", " + op2 + ", \"\", " +
+           to_str(I->isExact()) + ")";
+  } break;
+  case Instruction::FDiv: {
+    call = getFPBinArithOp("FDiv", op1, op2, I);
+  } break;
+  case Instruction::URem: {
+    call = "CreateURem(" + op1 + ", " + op2 + ")";
+  } break;
+  case Instruction::SRem: {
+    call = "CreateSRem(" + op1 + ", " + op2 + ")";
+  } break;
+  case Instruction::FRem: {
+    call = getFPBinArithOp("FRem", op1, op2, I);
+  } break;
+  case Instruction::Shl: {
+    call = getBinArithOp("Shl", op1, op2, I);
+  } break;
+  case Instruction::LShr: {
+    call = "CreateLShr(" + op1 + ", " + op2 + ", \"\", " +
+           to_str(I->isExact()) + ")";
+  } break;
+  case Instruction::AShr: {
+    call = "CreateAShr(" + op1 + ", " + op2 + ", \"\", " +
+           to_str(I->isExact()) + ")";
+  } break;
+  case Instruction::And: {
+    call = "CreateAnd(" + op1 + ", " + op2 + ")";
+  } break;
+  case Instruction::Or: {
+    call = "CreateOr(" + op1 + ", " + op2 + ")";
+  } break;
+  case Instruction::Xor: {
+    call = "CreateXor(" + op1 + ", " + op2 + ")";
+  } break;
+  case Instruction::Alloca: {
+    auto alI = dyn_cast<AllocaInst>(I);
+    auto val = alI->getArraySize();
+    auto valStr = val ? getVal(val) : "nullptr";
+    call = "CreateAlloca(" + getType(alI->getAllocatedType()) + ", " +
+           std::to_string(alI->getAddressSpace()) + ", " + valStr + ")";
+  } break;
+  case Instruction::Load: {
+    auto lI = dyn_cast<LoadInst>(I);
+    call = "CreateLoad(" + getType(I->getType()) + ", " + op1 + ", " +
+           to_str(lI->isVolatile()) + ")";
+  } break;
+  case Instruction::Store: {
+    auto sI = dyn_cast<StoreInst>(I);
+    call = "CreateStore(" + op1 + ", " + op2 + ", " + to_str(sI->isVolatile()) +
+           ")";
+  } break;
+  case Instruction::GetElementPtr: {
+    auto gepI = dyn_cast<GetElementPtrInst>(I);
+    std::string strList = llvmPrefix + "ArrayRef<" + llvmPrefix + "Value*>(";
+    if (I->getNumOperands() > 2)
+      strList += "{";
+    bool first = true;
+    for (unsigned i = 1; i < I->getNumOperands(); ++i) {
+      if (!first)
+        strList += ", ";
+      strList += getVal(I->getOperand(i));
+      first = false;
+    }
+    if (I->getNumOperands() > 2)
+      strList += "}";
+    strList += ")";
+    // TODO: For some structs the return type is ptr and it fails
+    call = "CreateGEP(" + getType(gepI->getResultElementType()) + ", " + op1 +
```

Comment:
```suggestion
    call = "CreateGEP(" + getType(gepI->getSourceElementType()) + ", " + op1 +
```
...would probably fix your GEP troubles.

Reply:
Thank you so much, that indeed was the issue.